### PR TITLE
refactor: accent colors

### PR DESCRIPTION
**Pull request by Schwehn42**
## Description

This PR refactors the accent colors. It's basically the second part to #4283, which changed all the color definitions.  

### Before
Previously, the two primary colors (light/dark) for each color was used to define the accent colors. For instance, using the classname `accent-color__backlog-blue` would yield the accent color `--accent-color: $blue--500` (and `accent-color--dark: $blue--400` respectively). On top of that, further CSS variables like `--accent-color-desaturated-hover-dark` were defined. The definitions were completely arbitrary and were used at own discretion instead of the actual design colors. 

Furthermore, some designs use accent colors beyond 500 and 400, e.g. the new primary/secondary buttons. This actually motivated me to refactor this.

### After
Now, not only the primary colors are used as accent colors, but all colors from the new schema are available. (e.g. `--accent-color--300`)
The primary colors are still easily accessible using shortcuts (`--accent-color--light`, `--accent-color-dark`).
Also, all shades are available as rgb, which means alpha channel can be applied.
"Excess" variables like the above mentioned desatured.. have been replaced with the new ones.

## Changelog

- move colors to separate file
- add colors to map
- new mixins `set-accent-variables` and `generate-accent-classes`
  - generate accent variables for each color class (like `backlog-blue`)
  - shades from `--accent-color--100` to `--accent-color-800`
  - shortcuts `--accent-color--light` and `--accent-color--dark`
  - split rgb version for each shade
  - remove excess accent variables
- use new accent colors in every file that requires them 
- some manual adjustements (e.g. concerning note reactions) to fix some colors

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The light- and dark-theme are both supported and tested
- [x] The design was implemented and is responsive for all devices and screen sizes
- [x] The application was tested in the most commonly used browsers (e.g. Chrome, Firefox, Safari)

### CODE REVIEW

File: src/components/ScrumlrLogo/ScrumlrLogo.scss [hunk 1]
```diff
@@ -39,7 +39,7 @@
   }
 
   .scrumlr-logo__stop {
-    stop-color: var(--accent-color);
+    stop-color: var(--accent-color--light);
```
Comment (BenedictHomuth) [inline src/components/ScrumlrLogo/ScrumlrLogo.scss:42]: The logo seems broken.
image

Comment (Schwehn42) [reply src/components/ScrumlrLogo/ScrumlrLogo.scss:42]: Oops I meant #4310 of course!

Comment (Schwehn42) [reply src/components/ScrumlrLogo/ScrumlrLogo.scss:42]: Should be fixed with #4312 now

**Review by BenedictHomuth — APPROVED** [review]
lgtm

File: src/components/ScrumlrLogo/ScrumlrLogo.scss (hunk 1 above)
Comment (BenedictHomuth) [reply src/components/ScrumlrLogo/ScrumlrLogo.scss:42]: nice

**Pull request merged**